WooCommerce/Classes/POS/Presentation/CartView.swift

```diff
@@ -2,6 +2,8 @@ import SwiftUI
 import protocol Yosemite.POSItem
 
 struct CartView: View {
+    @EnvironmentObject private var posModel: PointOfSaleAggregateModel
```

Comment:
Just curious: Is the benefit of using `@EnvironmentObject` so that a dependency doesn't have to be DI'ed manually? Might we run into the case where an environment object was forgotten and could it result in a crash?

Reply:
> Might we run into the case where an environment object was forgotten and could it result in a crash?

Just chiming in: Doesn't happen in this case, but from previous experiments I've done, if we forget to pass it we would get a `Fatal error: No ObservableObject of type SomeService found. A View.environmentObject(_:) for SomeService may be missing as an ancestor of this view.` In general we only have to ensure that whatever dependency we pass is initialized and injected into the environment on a higher-level on the hierarchy, then provided as an `@EnvironmentObject` in the root view and in the subviews that are needed.

Reply:
Yes, it makes it more convenient to use, especially when dealing with views deeper in the heirarchy and also with previews. 

It would indeed crash if we didn't include it... but it's a one-time include at the top level, and the app would crash 100% of the time when opening POS if we deleted that, so it seems low risk compared to the reward, at least to me.

Reply:
Nice, thanks for the clarifications. It's nice being able to pass a dependency for the whole view hierarchy. When we're ready for public release, maybe we can add a UI test to open POS just as a way to automate this check.